spec/unit/messages/service_instance_update_user_provided_message_spec.rb

```diff
@@ -99,9 +99,21 @@ module VCAP::CloudController
     end
 
     describe '.audit_hash' do
-      it 'produces a redacted audit hash' do
-        expected = body.dup.tap { |b| b[:credentials] = '[PRIVATE DATA HIDDEN]' }
-        expect(message.audit_hash).to match(expected.with_indifferent_access)
+      context 'when credentials are passed in' do
+        it 'produces a redacted audit hash' do
+          expected = body.dup.tap { |b| b[:credentials] = '[PRIVATE DATA HIDDEN]' }
```

Comment:
I tend to avoid doing this kind of things in tests as It is almost as duplicating the real code. I prefer to just build the hash manually.
Anyways, this is very simple so happy with it.

Reply:
Ya, not the biggest fan of this either, but I was just keeping the existing test mostly unchanged and just wrapping it in a context to mirror the case where no credentials are passed.